compiler/src/main/scala/org/scalajs/core/compiler/JSGlobalAddons.scala

```diff
@@ -188,7 +209,7 @@ trait JSGlobalAddons extends JSDefinitions
       assert(sym.isModuleClass,
           s"compat068FullJSNameOf called for non-module-class symbol $sym")
       sym.getAnnotation(JSFullNameAnnotation).flatMap(_.stringArg(0)) getOrElse {
-        jsNameOf(sym)
+        jsNameOf(sym).asInstanceOf[JSName.Literal].name
```

Comment:
Maybe a comment explaining why we can cast to `JSName.Literal` here? Basically because symbols did not exist in 0.6.8, so obviously if we're reading something from 0.6.8 it has to be a string.